lib/rig/event_stream/kafka_to_filter.ex

```diff
@@ -6,25 +6,24 @@ defmodule Rig.EventStream.KafkaToFilter do
   use Rig.KafkaConsumerSetup
 
   alias Rig.EventFilter
-  alias RigCloudEvents.CloudEvent
 
   # ---
 
   def validate(conf), do: {:ok, conf}
 
   # ---
 
-  def kafka_handler(message) do
-    case CloudEvent.parse(message) do
-      {:ok, %CloudEvent{} = cloud_event} ->
-        Logger.debug(fn -> inspect(cloud_event.parsed) end)
+  def kafka_handler(body, headers) do
+    case Cloudevents.from_kafka_message(body, headers) do
+      {:ok, cloud_event} ->
+        Logger.debug(fn -> inspect(cloud_event) end)
         EventFilter.forward_event(cloud_event)
         :ok
 
-      {:error, :parse_error} ->
-        {:error, :non_cloud_events_not_supported, message}
+      {:error, _reason} ->
+        {:error, :non_cloud_events_not_supported, body}
```

Comment:
including the `reason` in the `:error` tuple would make it visible in logs, so I suggest changing this to

```elixir
{:error, {:non_cloud_events_not_supported, reason, body}}
```